### ISSUE TITLE
Diagnoser crashing with 'undefined method `each` for nilclass'

### ISSUE THREAD

**Issue by taylorthurlow** [castwide/solargraph#337]
On Solargraph 0.39.10, I don't believe this was happening a few versions ago. Will downgrade and try to replicate.

```
project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/parser/legacy/node_methods.rb:127:

in `call_nodes_from': undefined method `each' for nil:NilClass (NoMethodError)

from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/parser/legacy/node_methods.rb:140:in `block in call_nodes_from'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/parser/legacy/node_methods.rb:140:in `each'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/parser/legacy/node_methods.rb:140:in `call_nodes_from'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/parser/legacy/node_methods.rb:140:in `block in call_nodes_from'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/parser/legacy/node_methods.rb:140:in `each'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/parser/legacy/node_methods.rb:140:in `call_nodes_from'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/parser/legacy/node_methods.rb:140:in `block in call_nodes_from'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/parser/legacy/node_methods.rb:140:in `each'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/parser/legacy/node_methods.rb:140:in `call_nodes_from'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/parser/legacy/node_methods.rb:140:in `block in call_nodes_from'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/parser/legacy/node_methods.rb:140:in `each'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/parser/legacy/node_methods.rb:140:in `call_nodes_from'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/type_checker.rb:210:in `call_problems'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/type_checker.rb:46:in `problems'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/diagnostics/type_check.rb:15:in `diagnose'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/library.rb:328:in `block in diagnose'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/library.rb:327:in `each_pair'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/library.rb:328:in `diagnose'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/language_server/host.rb:198:in `diagnose'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/language_server/host/diagnoser.rb:65:in `tick'
from project-path/vendor/ruby/2.5.0/gems/solargraph-0.39.10/lib/solargraph/language_server/host/diagnoser.rb:47:in `block in start'

    
```

**Comment by taylorthurlow** [comment]
Downgrading to 0.39.7 fixed my issue. Just a cursory guess based on the error and the changes to my gemfile when downgrading, maybe something to do with markdown parsing?

**Comment by castwide** [comment]
> EDIT: I _think_ it has something to do with the Rails helper comments usually suggested for better Rails integration. I've removed them for now.

That could be. I'm working on a possibly related bug in #329. Different error, but maybe the same root cause.

**Comment by castwide** [comment]
I haven't been able to reproduce this particular error, but the possibly related #329 bug is fixed in v0.39.11. Please let me know if the problem persists.

**Comment by castwide** [comment]
This bug still affected Ruby < 2.6. A fix has been released in v0.39.12.
